cupyx/scipy/sparse/util.py

```diff
@@ -22,3 +22,16 @@ def isshape(x):
         return False
     m, n = x
     return isintlike(m) and isintlike(n)
+
+
+def prune_array(array):
```

Comment:
If another array is a view of the large base array, the large memory space is not released and an additional memory area for the copy is newly allocated.

Reply:
@asi1024, if a copy is made, wouldn't the large base array be released as soon as control is returned to `prune()`, since it overwrites `self.indices` and `self.data` with the copies?

Reply:
I can change this however you prefer. Please let me know what I need to change.